easycv/predictors/classifier.py

```diff
@@ -122,3 +143,148 @@ def predict(self, input_data_list, batch_size=-1):
                 outputs_list.append(single_result)
 
         return outputs_list
+
+
+@PREDICTORS.register_module()
+class ClsPredictor(PredictorInterface):
+    """Inference image(s) with the detector.
+    Args:
+        model_path (str): checkpoint model and export model are shared.
+        config_path (str): If config_path is specified, both checkpoint model and export model can be used; if config_path=None, the export model is used by default.
+    """
+
+    def __init__(self, model_path, config_path=None):
+
+        self.model_path = model_path
+
+        if config_path is not None:
+            self.cfg = mmcv_config_fromfile(config_path)
+        else:
+            logger = get_root_logger()
+            logger.warning('please use export model!')
+            if is_url_path(self.model_path) and url_path_exists(
+                    self.model_path):
+                checkpoint = load_state_dict_from_url(model_path)
+            else:
+                assert io.exists(
+                    self.model_path), f'{self.model_path} does not exists'
+
+                with io.open(self.model_path, 'rb') as infile:
+                    checkpoint = torch.load(infile, map_location='cpu')
+
+            assert 'meta' in checkpoint and 'config' in checkpoint[
+                'meta'], 'meta.config is missing from checkpoint'
+
+            config_str = checkpoint['meta']['config']
+            if isinstance(config_str, dict):
+                config_str = json.dumps(config_str)
+
+            # get config
+            basename = os.path.basename(self.model_path)
+            fname, _ = os.path.splitext(basename)
+            self.local_config_file = os.path.join(CACHE_DIR,
+                                                  f'{fname}_config.json')
+            if not os.path.exists(CACHE_DIR):
+                os.makedirs(CACHE_DIR)
+            with open(self.local_config_file, 'w') as ofile:
+                ofile.write(config_str)
+            self.cfg = mmcv_config_fromfile(self.local_config_file)
+
+        # dynamic adapt mmdet models
+        dynamic_adapt_for_mmlab(self.cfg)
```

Comment:
why should we adapt for mmalb in classifier?

Reply:
done